exposed-migration/src/main/kotlin/MigrationUtils.kt

```diff
@@ -90,6 +91,48 @@ object MigrationUtils {
         return allStatements
     }
 
+    /**
+     * Returns the SQL statements that drop any columns that exist in the database but are not defined in [tables].
+     *
+     * By default, a description for each intermediate step, as well as its execution time, is logged at the INFO level.
+     * This can be disabled by setting [withLogs] to `false`.
+     *
+     * **Note:** Some dialects, like SQLite, do not support `ALTER TABLE DROP COLUMN` syntax completely.
+     * Please check the documentation.
+     */
+    fun dropUnmappedColumnsStatements(vararg tables: Table, withLogs: Boolean = true): List<String> {
+        if (tables.isEmpty()) return emptyList()
+
+        val statements = mutableListOf<String>()
+
+        val dbSupportsAlterTableWithDropColumn = TransactionManager.current().db.supportsAlterTableWithDropColumn
+
+        if (dbSupportsAlterTableWithDropColumn) {
+            val existingTablesColumns = logTimeSpent("Extracting table columns", withLogs) {
+                currentDialect.tableColumns(*tables)
+            }
```

Comment:
This is also always called in `addMissingColumnsStatements()` directly before with the same arguments, right? It may be inefficient to do this twice given the potential queries involved. I get that it must be possible in isolation because you've made this method public, so the metadata needs to be retrieved if this is invoked alone.

But maybe you could log an issue for us to later restructure the calls inside `statementsRequiredForDatabaseMigration()`? As in maybe we could have a single invokation to retrieve and store `existingTablesColumns` there. Then we could either have private overloads for `addMissingColumnsStatements` and `dropUnmappedColumnsStatements` or just default parameters that allow us to pass in the column data results & avoid duplicate queries.